nexus/src/app/disk.rs

```diff
@@ -510,6 +510,9 @@ impl super::Nexus {
             .project_delete_snapshot(opctx, &authz_snapshot, &db_snapshot)
             .await?;
 
+        // Kick off volume deletion saga
+        self.volume_delete(db_snapshot.volume_id).await?;
```

Comment:
This is what I was mentioning earlier - I 100% understand that deleting a snapshot should release a reference to a volume, but other things may also reference that volume, no? Shouldn't this be only actually performing the volume deletion if the volume isn't referenced by anything else?

Reply:
Okay, follow-up, I see I was being a dummy - this actually does exactly that, just in the saga.

We might want to mention - either via naming or a comment - that this isn't "unconditional volume deletion". It's really decrementing the volume refcount, which may or may not perform a deletion.

Reply:
excellent point, expanded the comment in 62e54166